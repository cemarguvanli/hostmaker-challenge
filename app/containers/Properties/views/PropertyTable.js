import React from 'react';
import PropTypes from 'prop-types';
import { upperFirst } from 'lodash';

import styles from './propertyTable.styles.scss';

class PropertyTable extends React.PureComponent {
  addressRenderer = (address) => (
    Object.keys(address).map((key, index) => (
      <p key={index}>{address[key]}</p> // eslint-disable-line
    ))
  )

  rowRenderer = (property, index) => (
    <tr key={index}>
      <td>{upperFirst(property.owner)}</td>
      <td>{this.addressRenderer(property.address)}</td>
      <td>Â£{property.incomeGenerated}</td>
    </tr>
  )

  render() {
    const { properties } = this.props;

    return (
      <div className={styles.tableWrapper}>
        <table className={styles.table}>
          <thead>
            <tr>
              <th>Owner</th>
              <th>Address</th>
              <th>IncomeGenerated</th>
            </tr>
          </thead>
          <tbody>
            {properties.map(this.rowRenderer)}
          </tbody>
        </table>
      </div>
    );
  }
}

PropertyTable.propTypes = {
  properties: PropTypes.array.isRequired,
};

export default PropertyTable;
