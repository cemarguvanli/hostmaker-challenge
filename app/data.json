[
  {
    "owner": "carlos",
    "address": {
        "line1": "Flat 5",
        "line4": "7 Westbourne Terrace",
        "postCode": "W2 3UL",
        "city": "London",
        "country": "U.K."
    },
    "airbnbId": 3512500,
    "numberOfBedrooms": 1,
    "numberOfBathrooms": 1,
    "incomeGenerated": 2000.34
  },
  {
    "owner": "ankur",
    "address": {
        "line1": "4",
        "line2": "Tower Mansions",
        "line3": "Off Station road",
        "line4": "86-87 Grange Road",
        "postCode": "SE1 3BW",
        "city": "London",
        "country": "U.K."
    },
    "airbnbId": 1334159,
    "numberOfBedrooms": 3,
    "numberOfBathrooms": 1,
    "incomeGenerated": 10000
  },
  {
    "owner": "elaine",
    "address": {
        "line1": "4",
        "line2": "332b",
        "line4": "Goswell Road",
        "postCode": "EC1V 7LQ",
        "city": "London",
        "country": "U.K."
    },
    "airbnbId": 12220057,
    "numberOfBedrooms": 2,
    "numberOfBathrooms": 2,
    "incomeGenerated": 1200
  }
]