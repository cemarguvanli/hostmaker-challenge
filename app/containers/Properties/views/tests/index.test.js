import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import data from '../../../../data.json';
import PropertyTable from '../PropertyTable';

describe('<PropertyTable />', () => {
  it('should render the properties table', () => {
    const wrapper = shallow(<PropertyTable properties={[]} />);
    expect(wrapper).toBeTruthy();
  });

  it('should render the properties table with 3 rows', () => {
    const wrapper = shallow(<PropertyTable properties={data} />);
    const rows = wrapper.find('tbody > tr');
    expect(rows.length).toEqual(3);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<PropertyTable properties={data} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // const rows = wrapper.find('.add-budget-table-row')
  //     expect(rows.length).to.eql(2)

  //     const firstRowColumns = rows.first().find('td').map(column => column.text())
  //     expect(firstRowColumns.length).to.eql(4)// since you have 4 td
  //     expect(firstRowColumns[0]).to.eql('BudgetCategoryName')
  //     expect(firstRowColumns[1]).to.eql(20)
  //     expect(firstRowColumns[2]).to.eql(someDate1)

// const SecondRowColumns = rows.last().find('td').map(column => column.text())
//       expect(firstRowColumns.length).to.eql(4)// since you have 4 td
//       expect(firstRowColumns[0]).to.eql('BudgetCategoryName2')
//       expect(firstRowColumns[1]).to.eql(30)
//       expect(firstRowColumns[2]).to.eql(someDate2)
});
