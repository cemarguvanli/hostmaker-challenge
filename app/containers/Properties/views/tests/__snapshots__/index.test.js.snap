// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PropertyTable /> renders correctly 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Owner
        </th>
        <th>
          Address
        </th>
        <th>
          IncomeGenerated
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Carlos
        </td>
        <td>
          <p>
            Flat 5
          </p>
          <p>
            7 Westbourne Terrace
          </p>
          <p>
            W2 3UL
          </p>
          <p>
            London
          </p>
          <p>
            U.K.
          </p>
        </td>
        <td>
          £
          2000.34
        </td>
      </tr>
      <tr>
        <td>
          Ankur
        </td>
        <td>
          <p>
            4
          </p>
          <p>
            Tower Mansions
          </p>
          <p>
            Off Station road
          </p>
          <p>
            86-87 Grange Road
          </p>
          <p>
            SE1 3BW
          </p>
          <p>
            London
          </p>
          <p>
            U.K.
          </p>
        </td>
        <td>
          £
          10000
        </td>
      </tr>
      <tr>
        <td>
          Elaine
        </td>
        <td>
          <p>
            4
          </p>
          <p>
            332b
          </p>
          <p>
            Goswell Road
          </p>
          <p>
            EC1V 7LQ
          </p>
          <p>
            London
          </p>
          <p>
            U.K.
          </p>
        </td>
        <td>
          £
          1200
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
